### ISSUE TITLE
.find() polyfill is needed

### ISSUE THREAD

**Issue by LoveMeWithoutAll** [angular-ui/ui-router#3725]
This issue tracker is for Bug Reports and Feature Requests only.

Please direct requests for help to StackOverflow.
See http://bit.ly/UIR-SOF for details.

This is a (check one box):

- [x] Bug Report
- [ ] Feature Request
- [ ] General Query 

My version of UI-Router is: 1.0.19(current stable lastest)

# Bug Report

#### Current Behavior:

I am using angular-ui-router with Angular1 and ocLazyLoad.
In Internet Explorer11, I am getting the below error message while instantiate module ui.router.

```javascript
SCRIPT5022: [$injector:modulerr] Failed to instantiate module majorApp due to:
Error: [$injector:modulerr] Failed to instantiate module ui.router due to:
Error: [$injector:modulerr] Failed to instantiate module ui.router.init due to:
TypeError: Object doesn't support property or method 'find'
   at getParamDeclaration (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:1603:9)
   at Param (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:1667:13)
   at ParamFactory.prototype.fromConfig (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:5044:13)
   at makeConfigParam (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:2732:59)
   at Anonymous function (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:515:50)
   at forEach (https://ajax.googleapis.com/ajax/libs/angularjs/1.7.2/angular.js:401:11)
   at map (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:515:9)
   at paramsBuilder (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:2734:13)
   at Anonymous function (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:2929:105)
   at StateBuilder.prototype.build (http://unpkg.com/@uirouter/angularjs/release/angular-ui-router.js:2930:17)
```

For solving this problem, I added the below polyfill script.

```html
<script src="//cdn.polyfill.io/v2/polyfill.min.js?features=default,MutationObserver"></script>
```

So it works.

#### Expected Behavior:

I think working without polyfill is needed as the last version v1.0.18.

#### Link to Plunker or stackblitz that reproduces the issue:

( if you want a response to your issue, provide a way to reproduce it )
( http://bit.ly/UIR-Plunk1 )
( https://stackblitz.com/edit/ui-router-angularjs )

# Feature Request

(type feature request here)

# General Query

Please direct general implementation questions to StackOverflow: 
http://stackoverflow.com/questions/ask?tags=angularjs,angular-ui-router

Please review the Sample Application which highlights common approaches:
https://github.com/ui-router/sample-app-angularjs

- [ ] I have already asked my question on StackOverflow and nobody could answer the question

- [x] I have already reviewed the sample application for examples of common approaches

- [x] I believe my question can only be answered by the UI-Router maintainers

(type general query here)

**Comment by LoveMeWithoutAll** [comment]
To resolve this error, polyfill is needed.

[This post](https://lovemewithoutall.github.io/it/angular-ui-router-needs-polyfill/) could be helpful

**Comment by ThomasStubbe** [comment]
+1

**Comment by brennaveen** [comment]
Your solution did not work for me.  I had to use this [polyfill ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Polyfill)from mozilla.org

**Comment by mattdsteele** [comment]
This has been fixed in core, but is waiting on ui-router/core@66a3244 to ship.

See ui-router/core#215

**Comment by LoveMeWithoutAll** [comment]
This problem was Fixed version 1.0.20
